### ISSUE TITLE
Model API

### ISSUE THREAD

**Issue by abhinavtripathy** [stocksmith/ml-research#7]
Parameters: Company Name, Time Scale, Weight)

**Comment by Saakshaat** [comment]
The Yahoo Finance API used to be the standard when it came to getting stock price data. It was depreciated sometime in 2017 but we could build something along similar lines. 

Most other publicly available APIs are paid. 
Here's an article for reference.
[https://towardsdatascience.com/best-5-free-stock-market-apis-in-2019-ad91dddec984](url)
@ApurvShah007 perhaps start by looking for the yfinance API and see how we can use it.

**Comment by Saakshaat** [comment]
Here are the parameters we should ideally be looking at when getting stock data:

- Stock Price (Open, Close, High, Low, Adj. Close[most important])

- Timeline

- Volume

- Ticker (Company Name)

These are the standard parameters listed in any stock data csv you scrape from any source.

**Comment by ApurvShah007** [comment]
@abhinavtripathy @Saakshaat This can easily be done by 2 libraries that I know and have worked with. yfinance and pandas_datareader are 2 very good libraries and easy to work with and provide the latest dat and the historic data. This should not be an issue.

**Comment by Saakshaat** [comment]
@ApurvShah007 since we're focusing on building packages, I'm not sure if we should be concerned with using an API since we won't be dealing with the data ourselves. We'll simply be providing the client (dev) with the functions, to which they send their data.

**Comment by ApurvShah007** [comment]
@Saakshaat Yes I remember we discussed that. That will call for some changes in the functions I have written and I will get on that asap.
